types/audit-details.d.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+declare global {
+  module LH.Audit.Details {
+    export interface Filmstrip {
+      type: 'filmstrip';
+      scale: number;
+      items: {
+        timing: number;
+        timestamp: number;
+        data: string;
+      }[];
+    }
+
+    export interface Screenshot {
+      type: 'screenshot';
+      timestamp: number;
+      data: string;
+    }
+
+    export interface Opportunity {
+      type: 'opportunity';
+      overallSavingsMs: number;
+      overallSavingsBytes?: number;
+      headings: TableColumnHeading[];
+      items: OpportunityItem[];
+    }
+
+    export interface CriticalRequestChain {
+      type: 'criticalrequestchain';
+      longestChain: {
+        duration: number;
+        length: number;
+        transferSize: number;
+      };
+      chains: Audit.SimpleCriticalRequestNode;
+    }
+
+    // TODO(bckenny)
+    // export interface MultiCheck {}
+    // export interface Table {}
+    // export interface SomeKindOfHiddenThing
+
+    // Contents of details below here
```

Comment:
maybe we want a different namespace here, not sure. These are things that are only inside details (table headings, table items).

Reply:
`LH.Audit.Details.Details` ;)

I'd personally be in favor of naming convention over a different namespace. they categorically they definitely belong under "details". maybe we return to `*Details` suffix if these get out of hand?

Reply:
> I'd personally be in favor of naming convention over a different namespace. they categorically they definitely belong under "details". maybe we return to `*Details` suffix if these get out of hand?

yeah, I'm ok with living with this for now, just trying to think of ways to shrink the amount of thinking one has to do. Part of this was motivated by reviewing #6999 and seeing some of our existing types like `DetailsRendererDetailsJSON` and `DetailsRendererNodeDetailsJSON` and going "wat" :)